In [1]:
!git branch

  JamBranch
* exploringTMDBdata
  master


In [2]:
import pandas as pd
import numpy as np

In [4]:
df = pd.read_csv('Data/tmdb_api_data.csv')
print(df.shape)
df.head()

(59429, 10)


Unnamed: 0   tmdb_id    imdb_id               title  \
0           0   44155.0  tt1612608         The Italian   
1           1  642129.0  tt1718842  Nakalnya Anak Muda   
2           2   69237.0  tt1645134          Sandcastle   
3           3   46249.0  tt1645129             Planzet   
4           4  347635.0  tt1645115               Naica   

                                        genre  \
0                                  ['Comedy']   
1             ['Horror', 'Drama', 'Thriller']   
2                                   ['Drama']   
3  ['Animation', 'Action', 'Science Fiction']   
4                                          []   

                                       studio  budget  revenue  tmdb_rating  \
0  ['Eskwad', 'Pathé!', 'M6 Films', 'Canal+']     0.0      0.0          5.3   
1                          ['Mitra Pictures']     0.0      0.0          0.0   
2                                          []     0.0      0.0          6.0   
3       ['CoMix Wave Films', 'Media Factory']     0.0      0.0          6.2   
4                                          []     0.0      0.0          0.0   

   tmdb_num_votes  
0           144.0  
1             0.0  
2             4.0  
3             5.0  
4             0.0

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 59429 entries, 0 to 59428
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Unnamed: 0      59429 non-null  int64  
 1   tmdb_id         59424 non-null  float64
 2   imdb_id         59424 non-null  object 
 3   title           59424 non-null  object 
 4   genre           59424 non-null  object 
 5   studio          59424 non-null  object 
 6   budget          59424 non-null  float64
 7   revenue         59424 non-null  float64
 8   tmdb_rating     59424 non-null  float64
 9   tmdb_num_votes  59424 non-null  float64
dtypes: float64(5), int64(1), object(4)
memory usage: 4.5+ MB


In [7]:
df.tmdb_id.isna().sum()

5

In [8]:
# There are 52,147 entries with a budget of 0
df.budget.value_counts()

0.0            52147
1000000.0        244
2000000.0        187
5000000.0        172
10000000.0       171
               ...  
260000000.0        1
7120949.0          1
693550.0           1
32679734.0         1
11500.0            1
Name: budget, Length: 1484, dtype: int64

In [11]:
# There are 55,233 movies with a revenue of 0
df.revenue.value_counts()

0.0           55233
10000.0          29
100000.0         25
1500000.0        17
2000000.0        13
              ...  
4500.0            1
13975.0           1
24426169.0        1
40462793.0        1
25574.0           1
Name: revenue, Length: 3759, dtype: int64

In [13]:
# 16,287 movies with a rating of 0
df.tmdb_rating.value_counts()

0.0    16287
6.0     3683
7.0     2909
5.0     2408
8.0     1758
       ...  
1.2        5
1.4        5
9.1        4
9.9        1
9.6        1
Name: tmdb_rating, Length: 93, dtype: int64

In [16]:
# 25,806 missing values
df.studio.value_counts()

[]                                                                                                                                25806
['SVF Entertainment']                                                                                                                76
['Star Cinema – ABS-CBN Film Productions']                                                                                           70
['Toho Company, Ltd.']                                                                                                               66
['Toei']                                                                                                                             57
                                                                                                                                  ...  
['BIG Curtains Media', 'NJ Lalwani Films']                                                                                            1
['Twisted Pictures', 'Protocol Entertainment', '

In [20]:
df[(df['budget']!=0)&(df['revenue']!=0)].shape

(2695, 10)

In [21]:
df.title.nunique()

56422

## Joining The Numbers dataset to fill missing values for budget and revenue

In [23]:
df = df.dropna()

In [28]:
numbers = pd.read_csv('Cleaned Data/cleaned_tn-movie-budgets.csv')
numbers.head()

Unnamed: 0  id  year             movie  production_budget  domestic_gross  \
0        5288  89  2000      The Specials            1000000           12996   
1        3308   9  2000         The Claim           13000000          622023   
2        1863  64  2000  Rugrats in Paris           30000000        76501438   
3         324  25  2000       The Patriot          110000000       113330342   
4        3891  92  2000   American Psycho            8000000        15070285   

   total_gross  profitability  
0        12996       0.012996  
1      1375635       0.105818  
2    103284813       3.442827  
3    215300000       1.957273  
4     34270285       4.283786

In [29]:
numbers.drop(['Unnamed: 0'], axis=1, inplace=True)

In [82]:
numbers.shape

(4387, 7)

In [81]:
numbers.total_gross.value_counts()

0             342
70958           1
116025023       1
406299          1
38329114        1
             ... 
426351163       1
1034727750      1
1518722794      1
402231063       1
38899792        1
Name: total_gross, Length: 4046, dtype: int64

In [87]:
df_merged = pd.merge(df, numbers, left_on='title',  right_on='movie', how='inner')
df_merged.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2451 entries, 0 to 2450
Data columns (total 17 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         2451 non-null   int64  
 1   tmdb_id            2451 non-null   float64
 2   imdb_id            2451 non-null   object 
 3   title              2451 non-null   object 
 4   genre              2451 non-null   object 
 5   studio             2451 non-null   object 
 6   budget             2451 non-null   float64
 7   revenue            2451 non-null   float64
 8   tmdb_rating        2451 non-null   float64
 9   tmdb_num_votes     2451 non-null   float64
 10  id                 2451 non-null   int64  
 11  year               2451 non-null   int64  
 12  movie              2451 non-null   object 
 13  production_budget  2451 non-null   int64  
 14  domestic_gross     2451 non-null   int64  
 15  total_gross        2451 non-null   int64  
 16  profitability      2451 

In [88]:
df_merged.drop(['budget', 'revenue', 'id', 'year', 'movie'], axis=1, inplace=True)

In [89]:
df_merged.drop(['Unnamed: 0'], axis=1, inplace=True)

In [90]:
df_merged.head()

tmdb_id    imdb_id                title  \
0   44639.0  tt1645089           Inside Job   
1   59015.0  tt1646127           Lights Out   
2  345911.0  tt4786282           Lights Out   
3   77934.0  tt1640714  The Christmas Bunny   
4  510578.0  tt1627937          The Journey   

                                        genre  \
0                             ['Documentary']   
1            ['Drama', 'Mystery', 'Thriller']   
2  ['Thriller', 'Mystery', 'Horror', 'Drama']   
3                         ['Drama', 'Family']   
4                                  ['Family']   

                                              studio  tmdb_rating  \
0                                  ['Sony Pictures']          7.7   
1  ['Diaphana Films', 'Région Ile-de-France', 'CN...          6.2   
2  ['Atomic Monster', 'New Line Productions', 'Ma...          6.3   
3                           ['Honey Creek Pictures']          5.9   
4  ['QOA Entertainment', 'DesertRock Entertainment']          0.0   

   tmdb_num_votes  production_budget  domestic_gross  total_gross  \
0           586.0            2000000         4311834      7799257   
1            32.0            5000000        67268835    148806510   
2          2747.0            5000000        67268835    148806510   
3            14.0             250000               0            0   
4             0.0             500000           19800        19800   

   profitability  
0       3.899629  
1      29.761302  
2      29.761302  
3       0.000000  
4       0.039600

In [140]:
df_merged.domestic_gross.value_counts()

0            367
177397510     10
15433         10
656688         8
28544157       8
            ... 
8110621        1
246305         1
85710210       1
229024295      1
10219501       1
Name: domestic_gross, Length: 1687, dtype: int64

In [98]:
df_merged_zero_rev = df_merged[df_merged['total_gross']==0]
df_merged_zero_rev.shape

(252, 11)

There are 252 missing values for total gross in The numbers dataset, so we're going to see if we can fill in any of these values with the BOM dataset.

In [92]:
bom_dataset = pd.read_csv('Cleaned Data/cleaned_bom-movie-gross.csv')
bom_dataset.head()

Unnamed: 0                                        title studio  \
0           0                                  Toy Story 3     BV   
1           1                   Alice in Wonderland (2010)     BV   
2           2  Harry Potter and the Deathly Hallows Part 1     WB   
3           3                                    Inception     WB   
4           4                          Shrek Forever After   P/DW   

   domestic_gross  foreign_gross  year   total_gross  
0     415000000.0    652000000.0  2010  1.067000e+09  
1     334200000.0    691300000.0  2010  1.025500e+09  
2     296000000.0    664300000.0  2010  9.603000e+08  
3     292600000.0    535700000.0  2010  8.283000e+08  
4     238700000.0    513900000.0  2010  7.526000e+08

In [99]:
bom_numbers = pd.merge(df_merged_zero_rev, bom_dataset, on='title', how='left')
print(bom_numbers.shape)
bom_numbers.head(3)

(252, 17)


tmdb_id    imdb_id                title                            genre  \
0   77934.0  tt1640714  The Christmas Bunny              ['Drama', 'Family']   
1  272795.0  tt1675846            Checkmate             ['Mystery', 'Drama']   
2  356841.0  tt3781616            Checkmate  ['Thriller', 'Action', 'Crime']   

                                            studio_x  tmdb_rating  \
0                           ['Honey Creek Pictures']          5.9   
1                                                 []          0.0   
2  ['Status Media & Entertainment', 'Puppy Entert...          4.0   

   tmdb_num_votes  production_budget  domestic_gross_x  total_gross_x  \
0            14.0             250000                 0              0   
1             0.0            1500000                 0              0   
2            21.0            1500000                 0              0   

   profitability  Unnamed: 0 studio_y  domestic_gross_y  foreign_gross  year  \
0            0.0         NaN      NaN               NaN            NaN   NaN   
1            0.0         NaN      NaN               NaN            NaN   NaN   
2            0.0         NaN      NaN               NaN            NaN   NaN   

   total_gross_y  
0            NaN  
1            NaN  
2            NaN

In [100]:
bom_numbers.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 252 entries, 0 to 251
Data columns (total 17 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   tmdb_id            252 non-null    float64
 1   imdb_id            252 non-null    object 
 2   title              252 non-null    object 
 3   genre              252 non-null    object 
 4   studio_x           252 non-null    object 
 5   tmdb_rating        252 non-null    float64
 6   tmdb_num_votes     252 non-null    float64
 7   production_budget  252 non-null    int64  
 8   domestic_gross_x   252 non-null    int64  
 9   total_gross_x      252 non-null    int64  
 10  profitability      252 non-null    float64
 11  Unnamed: 0         11 non-null     float64
 12  studio_y           11 non-null     object 
 13  domestic_gross_y   11 non-null     float64
 14  foreign_gross      11 non-null     float64
 15  year               11 non-null     float64
 16  total_gross_y      11 non-

In [102]:
bom_numbers.dropna(inplace=True)

In [104]:
bom_numbers

tmdb_id    imdb_id   title                                      genre  \
39   587506.0  tt1111235  Trance                                 ['Horror']   
40    68727.0  tt1924429  Trance  ['Thriller', 'Crime', 'Drama', 'Mystery']   
65   295180.0  tt1663708  Snitch                                         []   
66   134411.0  tt0882977  Snitch            ['Thriller', 'Drama', 'Action']   
94    96599.0  tt1734433    Eden                                  ['Drama']   
95   216337.0  tt2063641    Eden                                         []   
96   137530.0  tt2362758    Eden                                  ['Drama']   
97   249929.0  tt2959502    Eden                                  ['Drama']   
98   283330.0  tt3090634    Eden                         ['Drama', 'Music']   
99   360339.0  tt3032282    Eden                      ['Thriller', 'Drama']   
100  587829.0  tt9174880    Eden                                  ['Drama']   

                                              studio_x  tmdb_rating  \
39   ['Louisiana Media Productions', 'Skinny Giraff...          0.0   
40   ['Pathé!', 'Decibel Films', 'Film4 Productions...          6.6   
65                                                  []          1.0   
66   ['Exclusive Media', 'Participant Media', 'Imag...          6.0   
94           ['Centripetal Films', 'Eden Productions']          6.7   
95                                                  []          0.0   
96                                   ['Daza Cultural']          6.5   
97   ['Gabriel Figueroa Film fund', 'Atlantis 8 Fil...          0.0   
98   ['Ad Vitim Distribution', 'C.G. Cinema', 'Blue...          5.9   
99   ['Ombra Films', 'Perlapartment Pictures', 'Gob...          5.4   
100                                                 []          0.0   

     tmdb_num_votes  production_budget  domestic_gross_x  total_gross_x  \
39              0.0             950000                 0              0   
40           1550.0             950000                 0              0   
65              1.0             850000                 0              0   
66           1751.0             850000                 0              0   
94            133.0            2300000                 0              0   
95              0.0            2300000                 0              0   
96              2.0            2300000                 0              0   
97              0.0            2300000                 0              0   
98             75.0            2300000                 0              0   
99             32.0            2300000                 0              0   
100             0.0            2300000                 0              0   

     profitability  Unnamed: 0 studio_y  domestic_gross_y  foreign_gross  \
39             0.0      1262.0     FoxS         2300000.0     21900000.0   
40             0.0      1262.0     FoxS         2300000.0     21900000.0   
65             0.0      1245.0     LG/S        42900000.0            0.0   
66             0.0      1245.0     LG/S        42900000.0            0.0   
94             0.0      2225.0       BG           65500.0            0.0   
95             0.0      2225.0       BG           65500.0            0.0   
96             0.0      2225.0       BG           65500.0            0.0   
97             0.0      2225.0       BG           65500.0            0.0   
98             0.0      2225.0       BG           65500.0            0.0   
99             0.0      2225.0       BG           65500.0            0.0   
100            0.0      2225.0       BG           65500.0            0.0   

       year  total_gross_y  
39   2013.0     24200000.0  
40   2013.0     24200000.0  
65   2013.0     42900000.0  
66   2013.0     42900000.0  
94   2015.0        65500.0  
95   2015.0        65500.0  
96   2015.0        65500.0  
97   2015.0        65500.0  
98   2015.0        65500.0  
99   2015.0        65500.0  
100  2015.0        65500.0

In [114]:
bom_numbers.domestic_gross_x = bom_numbers.domestic_gross_y
bom_numbers.total_gross_x = bom_numbers.total_gross_y
bom_numbers.head()

tmdb_id    imdb_id   title                                      genre  \
39  587506.0  tt1111235  Trance                                 ['Horror']   
40   68727.0  tt1924429  Trance  ['Thriller', 'Crime', 'Drama', 'Mystery']   
65  295180.0  tt1663708  Snitch                                         []   
66  134411.0  tt0882977  Snitch            ['Thriller', 'Drama', 'Action']   
94   96599.0  tt1734433    Eden                                  ['Drama']   

                                             studio_x  tmdb_rating  \
39  ['Louisiana Media Productions', 'Skinny Giraff...          0.0   
40  ['Pathé!', 'Decibel Films', 'Film4 Productions...          6.6   
65                                                 []          1.0   
66  ['Exclusive Media', 'Participant Media', 'Imag...          6.0   
94          ['Centripetal Films', 'Eden Productions']          6.7   

    tmdb_num_votes  production_budget  domestic_gross_x  total_gross_x  \
39             0.0             950000         2300000.0     24200000.0   
40          1550.0             950000         2300000.0     24200000.0   
65             1.0             850000        42900000.0     42900000.0   
66          1751.0             850000        42900000.0     42900000.0   
94           133.0            2300000           65500.0        65500.0   

    profitability  Unnamed: 0 studio_y  domestic_gross_y  foreign_gross  \
39            0.0      1262.0     FoxS         2300000.0     21900000.0   
40            0.0      1262.0     FoxS         2300000.0     21900000.0   
65            0.0      1245.0     LG/S        42900000.0            0.0   
66            0.0      1245.0     LG/S        42900000.0            0.0   
94            0.0      2225.0       BG           65500.0            0.0   

      year  total_gross_y  
39  2013.0     24200000.0  
40  2013.0     24200000.0  
65  2013.0     42900000.0  
66  2013.0     42900000.0  
94  2015.0        65500.0

In [115]:
bom_numbers.drop(['Unnamed: 0', 'studio_y', 'domestic_gross_y', 'foreign_gross', 'year', 'total_gross_y'], axis=1, inplace=True)
bom_numbers.columns

Index(['tmdb_id', 'imdb_id', 'title', 'genre', 'studio_x', 'tmdb_rating',
       'tmdb_num_votes', 'production_budget', 'domestic_gross_x',
       'total_gross_x', 'profitability'],
      dtype='object')

In [120]:
bom_numbers = bom_numbers.rename(columns={'studio_x': 'studio', 'domestic_gross_x': 'domestic_gross', 'total_gross_x': 'total_gross'})
bom_numbers.columns

Index(['tmdb_id', 'imdb_id', 'title', 'genre', 'studio', 'tmdb_rating',
       'tmdb_num_votes', 'production_budget', 'domestic_gross', 'total_gross',
       'profitability'],
      dtype='object')

In [121]:
bom_numbers.shape

(11, 11)

In [127]:
bom_numbers.tmdb_id

39     587506.0
40      68727.0
65     295180.0
66     134411.0
94      96599.0
95     216337.0
96     137530.0
97     249929.0
98     283330.0
99     360339.0
100    587829.0
Name: tmdb_id, dtype: float64

We were able to fill in missing values for domestic gross revenue with values found in the BOM dataset. We were only able to do this for 11 of the 252 missing values. We'll now remove all cells where total gross revenue is equal to zero.

In [141]:
df_merged = df_merged[df_merged['total_gross']!=0]
df_merged.shape

(2199, 11)

In [143]:
df_final = pd.concat([df_merged, bom_numbers])
df_final.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2210 entries, 0 to 100
Data columns (total 11 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   tmdb_id            2210 non-null   float64
 1   imdb_id            2210 non-null   object 
 2   title              2210 non-null   object 
 3   genre              2210 non-null   object 
 4   studio             2210 non-null   object 
 5   tmdb_rating        2210 non-null   float64
 6   tmdb_num_votes     2210 non-null   float64
 7   production_budget  2210 non-null   int64  
 8   domestic_gross     2210 non-null   float64
 9   total_gross        2210 non-null   float64
 10  profitability      2210 non-null   float64
dtypes: float64(6), int64(1), object(4)
memory usage: 207.2+ KB


In [146]:
df_final.total_gross.value_counts()

385997896.0    10
44793168.0     10
17856688.0      8
5953886.0       8
45043870.0      8
               ..
677923379.0     1
115149422.0     1
128256712.0     1
18733666.0      1
3632.0          1
Name: total_gross, Length: 1772, dtype: int64

In [147]:
df_final.to_csv('Cleaned Data/cleaned_tmdb-num-bom.csv')

### What if we had merged with BOM instead of numbers?

In [107]:
df_merged_bom = pd.merge(df, bom_dataset, on='title', how='inner')
df_merged_bom.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2965 entries, 0 to 2964
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Unnamed: 0_x    2965 non-null   int64  
 1   tmdb_id         2965 non-null   float64
 2   imdb_id         2965 non-null   object 
 3   title           2965 non-null   object 
 4   genre           2965 non-null   object 
 5   studio_x        2965 non-null   object 
 6   budget          2965 non-null   float64
 7   revenue         2965 non-null   float64
 8   tmdb_rating     2965 non-null   float64
 9   tmdb_num_votes  2965 non-null   float64
 10  Unnamed: 0_y    2965 non-null   int64  
 11  studio_y        2965 non-null   object 
 12  domestic_gross  2965 non-null   float64
 13  foreign_gross   2965 non-null   float64
 14  year            2965 non-null   int64  
 15  total_gross     2965 non-null   float64
dtypes: float64(8), int64(3), object(5)
memory usage: 393.8+ KB


We chose against using BOM becasue it does not include prodction budget.